### PR TITLE
Fix duplicate user props

### PR DESCRIPTION
**Pull request by becca-bailey**
This fixes a bug that Paul and I were running into where the user props like `data-testid` and `aria-label` were being passed in to both the container and the group component, so even when we were rendering a victory data component as a standalone component (without a container or VictoryChart wrapper), `screen.getByTestId` was returning multiple nodes in the tests rather than the parent wrapper component.

In order to fix this, I removed the user props logic from the `renderData` function in `add-events`, and passed the user props into the root component that was returned from a Victory component rather than the container. This ensures that the props will be passed into either the `svg` or the `g` component, depending on whether there is another container present. I also added some additional test coverage around this, which we should add to other component tests as we convert them.

### CODE REVIEW

File: packages/victory-core/src/victory-clip-container/victory-clip-container.js
Comment (becca-bailey) [inline packages/victory-core/src/victory-clip-container/victory-clip-container.js:88]: I don't love that we have to explicitly call this here when we are also calling this function in the top-level component. However, there are props here that we can't pass into the base HTML component, so we do need some way of identifying which "extra" props are okay to pass in (using `...rest` creates DOM warnings). There is a bigger issue here, which is the issue where we are passing this props object around, and don't have visibility into what is in it.

File: test/jest/rendering-utils.js
Comment (becca-bailey) [inline test/jest/rendering-utils.js:5]: I had forgotten that there is a slightly easier way to add a custom wrapper component. Does it still make sense to have this helper, or is it just as clear if we just add `{ wrapper: "svg" }` to each individual test where we need it, rather than having to remember to use this helper?

File: test/jest/victory-area/victory-area.test.js
Comment (heythisispaul) [inline test/jest/victory-area/victory-area.test.js:12]: Good thinking adding this check onto each of these components 👍

File: packages/victory-core/src/victory-clip-container/victory-clip-container.js
Comment (heythisispaul) [reply packages/victory-core/src/victory-clip-container/victory-clip-container.js:88]: Reading through this PR really opened by eyes to how much "magic" is happening in Victory to get (the appropriate) props applied to children.

This whitelisting of props (`safeUserProps`) seems to be flexible enough, but I agree it leads to a lot of manual checks and only checks for known cases of "unsafe" props.

Are there ideas to handle this differently in a future major version?

**Comment by heythisispaul** [comment]
Other than maybe deleting that unneeded test util, this looks good to me!

File: test/jest/victory-area/victory-area.test.js
Comment (scottrippey) [inline test/jest/victory-area/victory-area.test.js:23]: Isn't `toHaveAttribute` part of `jest-dom`?  I'm confused how this is working :)

File: test/jest/victory-area/victory-area.test.js
Comment (scottrippey) [inline test/jest/victory-area/victory-area.test.js:40]: Should we test this?  This seems like an assertion on an implementation detail ... but then again, I guess you could just be validating that the attributes were added to the correct component

File: test/jest/victory-bar/victory-bar.test.js
Comment (scottrippey) [inline test/jest/victory-bar/victory-bar.test.js:9]: I don't see this added to `devDependencies` yet ... maybe it's a "transient" dependency already?  In my PR, I also configured it globally using `setupFilesBeforeEnv`, maybe you can cherry-pick that config?

**Review by scottrippey — APPROVED** [review]
Great changes, glad to see an easy way to add these userProps.
Minor changes requested for using `jest-dom`, otherwise good!

File: test/jest/victory-bar/victory-bar.test.js
Comment (scottrippey) [reply test/jest/victory-bar/victory-bar.test.js:9]: (you don't really have to cherry-pick, it's fine to just wait for my PR to merge)

File: packages/victory-core/src/victory-clip-container/victory-clip-container.js
Comment (becca-bailey) [reply packages/victory-core/src/victory-clip-container/victory-clip-container.js:88]: Yep, two ideas. First, I think we should take a bunch of the logic for providing prop defaults and getting derived values from the component's props and put it in a context wrapper or some sort of top-level state. That way, we have a single source of truth for accessing the original props, defaults, and derived values. Right now that's happening in many different places with functions like `getBaseProps`, `getCalculatedProps`, and all these individual render functions. 

Second, I want to add explicit types to these helper and be more explicit about what is being passed into them to make it all a little less mysterious.

File: test/jest/rendering-utils.js
Comment (becca-bailey) [reply test/jest/rendering-utils.js:5]: I think I will just remove this so we don't have to remember another import. FYI @scottrippey

File: test/jest/victory-area/victory-area.test.js
Comment (becca-bailey) [reply test/jest/victory-area/victory-area.test.js:23]: https://github.com/FormidableLabs/victory/blob/fa406575ba917fb2b84e2c6b31a5201fbd255b5d/test/jest/victory-area/victory-area.test.js#L1

File: test/jest/victory-bar/victory-bar.test.js
Comment (becca-bailey) [reply test/jest/victory-bar/victory-bar.test.js:9]: I'll just fix this once your PR gets merged.

File: packages/victory-core/src/victory-clip-container/victory-clip-container.js
Comment (heythisispaul) [reply packages/victory-core/src/victory-clip-container/victory-clip-container.js:88]: Top-level context was the first thing I was thinking of as well. It'd be easy enough to define an API for the props of the top level component and build a context value with that, or even maybe introduce something like a `VictoryProvider` or something where the user can do it themselves.

This is helpful, thanks!

**Review by heythisispaul — APPROVED** [review]

**Comment by becca-bailey** [comment]
@heythisispaul you should be able to merge this into your PR now.

**Pull request merged**
